packages/@react-types/textfield/src/index.d.ts

```diff
@@ -39,7 +39,8 @@ export interface AriaTextFieldProps extends TextFieldProps, AriaLabelingProps, F
    */
   'aria-autocomplete'?: 'none' | 'inline' | 'list' | 'both',
   /** Indicates the availability and type of interactive popup element, such as menu or dialog, that can be triggered by an element. */
-  'aria-haspopup'?: boolean | 'menu' | 'listbox' | 'tree' | 'grid' | 'dialog',
+
+  'aria-haspopup'?: boolean | 'false' | 'true' | 'menu' | 'listbox' | 'tree' | 'grid' | 'dialog',
```

Comment:
If I set this back, I'm not seeing any typescript errors, can we put it back to the way it was?

Reply:
ah I think you don't see errors because I set mergeProps to take and return any value: https://github.com/adobe/react-spectrum/pull/731/files#diff-4bbc55097af7fdcc45f7b7d721e3a94bR148

Otherwise, it was too conflicting because we merged multiple props with different roles. And almost all of them inherit their type props from other props which makes it hard to override them.

Reply:
We can fix the typing errors with some tricks using Typescript utility types like `Omit` and `Pick`

Reply:
basically, yes, we can put it back to the way it was since there'll be no type errors

Reply:
I tried to set the type back to valid types. And noticed that [`min`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/min), [`max`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/max) and [`step`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/step) attributes are valid for specific input types.
Since, we don't use type="number" anymore it would not make sense to expose those attributes I think

Reply:
I don't think you need to set https://github.com/adobe/react-spectrum/pull/731/files#diff-4bbc55097af7fdcc45f7b7d721e3a94bR148 to any, if I remove that, I still see no errors
As for min/max, I think now we should use aria-min/max/valuenow
@majornista thoughts on this?

To bring you up to speed @majornista, we're following the recommendation from https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/

Reply:
Following [gov.uk recommendation](https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/), where the `input` now has `type="text"`, I don't think that aria-valuemin/max/now will be valid attributes anymore. Another question I have is how to add the `[pattern]` attribute appropriately based on locale. A `input[type="number"]` displays values differently depending on the locale.

Reply:
It should have `role="spinbutton"` though, so aria-valuemin/max/now should work?

Reply:
aria attributes are set in `useSpinButton`
https://github.com/deebov/react-spectrum/blob/616810ebda2227051d0b5e0f4b70033234ca164f/packages/%40react-aria/spinbutton/src/useSpinButton.ts#L152

Reply:
> I don't think that aria-valuemin/max/now will be valid attributes anymore

@majornista [according to MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-valuemax_attribute) aria-valuemin/max/now don't depend on specific input types but are required to role="spinbutton"

I think we just need to remove the raw min/max/step attirbutes which are not valid for type="text"

Reply:
@deebov, That sounds right. But, `spinButtonProps` are not actually getting set to the text input at present.  I think they are getting left on the cutting room floor when we call `useTextField` at: https://github.com/adobe/react-spectrum/pull/731/files#diff-4bbc55097af7fdcc45f7b7d721e3a94bR148.

Reply:
Should we include support for `[inputMode]` and `[pattern]` as well? To support them we would need to inherit `TextInputDOMProps` into the `NumberFieldProps` interface at: https://github.com/adobe/react-spectrum/pull/731/files#diff-e824d1f31f734eeeddcffa299204f2f7R25.

Reply:
If it diverges enough from useTextField, should we consider NumberField to be it's own and not try for useTextField to do too much?